providers/flagd/pkg/provider.go

```diff
@@ -0,0 +1,161 @@
+package flagd
+
+import (
+	"github.com/open-feature/golang-sdk-contrib/providers/flagd/pkg/service"
+	GRPCService "github.com/open-feature/golang-sdk-contrib/providers/flagd/pkg/service/grpc"
+	HTTPService "github.com/open-feature/golang-sdk-contrib/providers/flagd/pkg/service/http"
+	of "github.com/open-feature/golang-sdk/pkg/openfeature"
+)
+
+type Provider struct {
+	service     service.IService
+	port        int32
+	host        string
+	serviceName string
+	protocol    string
+}
+
+type ProviderOption func(*Provider)
+
+func NewProvider(opts ...ProviderOption) *Provider {
+	provider := &Provider{
+		serviceName: "http",
+		port:        8080,
+		host:        "localhost",
+		protocol:    "http",
+	}
+	for _, opt := range opts {
+		opt(provider)
+	}
+	if provider.serviceName == "grpc" {
+		provider.service = GRPCService.NewGRPCService(
+			GRPCService.WithPort(provider.port),
+			GRPCService.WithHost(provider.host),
+		)
+	} else {
+		provider.service = HTTPService.NewHTTPService(
+			HTTPService.WithPort(provider.port),
+			HTTPService.WithHost(provider.host),
+			HTTPService.WithProtocol(provider.protocol),
+		)
+	}
+	return provider
+}
+
+func WithPort(port int32) ProviderOption {
+	return func(p *Provider) {
+		p.port = port
+	}
+}
+
+func WithHost(host string) ProviderOption {
+	return func(p *Provider) {
+		p.host = host
+	}
+}
+
+// service should be one of "http" or "grpc", if not the default "http" will be used
```

Comment:
```suggestion
// WithService specifies the type of the service. service should be one of "http" or "grpc", if not the default "http" will be used
```

to make the linters happy, comments of exported functions shall start with their name.
Alos, consider introducing kind of an enum `ServiceType` to improve ux, e.g:
```
type ServiceType string

const (
	HTTP ServiceType = "http"
	GRPC             = "grpc"
)
```